validate_binary_search_tree recursive.java

```diff
@@ -0,0 +1,56 @@
+// Time Complexity : O(n), where n is the number of nodes in the tree
+// Space Complexity :O(n), where n is the height of the tree, to maintain stack
+// Did this code successfully run on Leetcode : yes
+// Any problem you faced while coding this : not clear, if how function level return terminates the program, but base case level return 

```

Comment:
Its like 3(parent) calling 2 calling 1 (base). Return on base case terminates 1 thus control returns to 2. Now return in the recursive case terminates 2 and control returns to 3. The recursive case return terminates 3 and control transfers to driver or main. If still unclear, please DM